integration-tests/src/test/java/oracle/kubernetes/operator/utils/JRFDomain.java

```diff
@@ -0,0 +1,215 @@
+// Copyright 2019, Oracle Corporation and/or its affiliates.  All rights reserved.
+// Licensed under the Universal Permissive License v 1.0 as shown at
+// http://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.operator.utils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Objects;
+import oracle.kubernetes.operator.BaseTest;
+import org.yaml.snakeyaml.Yaml;
+
+/** JRF Domain class with all the utility methods */
+public class JRFDomain extends Domain {
+
+  private static final String FMWINFRA_DOCKER_IMAGENAME =
+      "phx.ocir.io/weblogick8s/oracle/fmw-infrastructure";
+  private static final String FMWINFRA_DOCKER_IMAGETAG = "12.2.1.3";
+  private static final String OCIR_REPO_SERVER = "phx.ocir.io";
+
+  /**
+   * JRFDomain constructor
+   *
+   * @param inputYaml - jrf domain input yaml file, which should contain the properties used for jrf
+   *     domain creation
+   * @throws Exception - if any error occurs
+   */
+  public JRFDomain(String inputYaml) throws Exception {
+    // read input domain yaml to test
+    this(TestUtils.loadYaml(inputYaml));
+  }
+
+  /**
+   * JRFDomain constructor
+   *
+   * @param inputDomainMap - jrf domain input properties map, which should contain the properties
+   *     used for domain creation
+   * @throws Exception - if any error occurs
+   */
+  public JRFDomain(Map<String, Object> inputDomainMap) throws Exception {
+    initialize(inputDomainMap);
+    createPV();
+    createSecret();
+    generateInputYaml();
+    callCreateDomainScript(userProjectsDir);
+    // TODO: add load balancer later
+    // createLoadBalancer();
+  }
+
+  /**
+   * initialize the domain creation input properties
+   *
+   * @param inputDomainMap - jrf domain input properties map
+   * @throws Exception - if any error occurs
+   */
+  private void initialize(Map<String, Object> inputDomainMap) throws Exception {
```

Comment:
There is lot of code duplication from Domain.java here. Its already out of sync with latest changes. Can you ping me to discuss how we can resolve this.

Reply:
I pushed another commit to remove the duplicates. Please review. Thanks!